src/callbacks/callback_print.cpp

```diff
@@ -193,7 +193,7 @@ void lbann_callback_print::report_results(model *m) {
         comm->intermodel_gather(score, score_list);
         comm->intermodel_gather(score_samples, num_samples_list);
         for (int i = 0; i < num_models; ++i) {
-          std::cout << "Model " << i << " " << mode_string << " "
+          std::cout << "Model " << i << " " << m->get_name() << " " << mode_string << " "
```

Comment:
Nothing of mine will break.

Reply:
We might already break the parse scripts since we insert an extra space. In any case, my formatting complaints are probably beyond the scope of this PR. I think the following will suffice for now:
```
std::cout << "Model " << i << " ";
if (!m->get_name().empty()) {
  std::cout << "(" << m->get_name() << ") ";
}
std::cout << mode_string << " ";
```

This won't affect the parse scripts. It's not especially clear, but we'll probably need a long discussion before we figure out good ways to make our output more legible.